modules/vmfcore/java/com/intel/vmf/MetadataStream.java

```diff
@@ -0,0 +1,435 @@
+    {
+        n_removeSet (nativeObj, set.nativeObj);
+    }
+    
+    public void remove (MetadataSchema schema)
+    {
+        n_removeSchema (nativeObj, schema.nativeObj);
+    }
+    
+    public void remove()
+    {
+        n_removeAllMd (nativeObj);
+    }
+    
+    public void addSchema (MetadataSchema schema)
+    {
+        n_addSchema (nativeObj, schema.nativeObj);
+    }
+    
+    public MetadataSchema getSchema (String schemaName)
+    {
+        return new MetadataSchema (n_getSchema (nativeObj, schemaName));
+    }
+    
+    public String[] getAllSchemaNames ()
+    {
+        return n_getAllSchemaNames (nativeObj);
+    }
+    
+    public MetadataSet getAll()
+    {
+        return new MetadataSet (n_getAll (nativeObj));
+    }
+    
+    public boolean importSet (MetadataStream srcStream, MetadataSet srcSet,
+                              long tarFrameIndex, long srcFrameIndex)
+    {
+        return n_importSet (nativeObj, srcStream.nativeObj, srcSet.nativeObj,
+                            tarFrameIndex, srcFrameIndex, FRAME_COUNT_ALL, 0);
+    }
+    
+    public boolean importSet (MetadataStream srcStream, MetadataSet srcSet, long tarFrameIndex, 
+                              long srcFrameIndex, long numOfFrames, MetadataSet setFailure)
+    {
+        return n_importSet (nativeObj, srcStream.nativeObj, srcSet.nativeObj,
+                            tarFrameIndex, srcFrameIndex, numOfFrames, setFailure.nativeObj);
+    }
+    
+    public void clear ()
+    {
+       //TO DO: implementation 
+    }
+    
+    public void sortById ()
+    {
+        n_sortMdSetById (nativeObj);
+    }
+    
+    public String serialize (IWriter formater)
+    {
+        return n_serialize (nativeObj, formater.getNativeAddr());
+    }
+    
+    public void deserialize (String text, IReader formater)
+    {
+        n_deserialize (nativeObj, text, formater.getNativeAddr());
+    }
+    
+    public String computeChecksum ()
+    {
+        return n_computeChecksum (nativeObj);
+    }
+    
+    public String computeChecksum (long XMPPacketSize, long XMPPacketOffset)
+    {
+        return n_computeChecksum (nativeObj, XMPPacketSize, XMPPacketOffset);
+    }
+    
+    public String getChecksum ()
+    {
+        return n_getChecksum (nativeObj);
+    }
+    
+    public void setChecksum (String checksum)
+    {
+        n_setChecksum (nativeObj, checksum);
+    }
+    
+    public void addVideoSegment (VideoSegment newSegment)
+    {
+        n_addVideoSegment (nativeObj, newSegment.nativeObj);
+    }
+    
+    public VideoSegment[] getAllVideoSegments ()
+    {
+        long nObjs[] = n_getAllVideoSegments (nativeObj);
+        VideoSegment objs[] = new VideoSegment[nObjs.length];
+        
+        for (int i = 0; i < nObjs.length; i++)
+        {
+            objs[i] = new VideoSegment (nObjs[i]);
+        }
+        
+        return objs;
+    }
+    
+    public long[] convertTimestampToFrameIndex (long timestamp, long duration,
```

Comment:
let's split it into `long convertTimestampToFrameIndex (long timestamp);` and
`long convertDurationToNumOfFrames(long timestamp, long duration);`